crates/resolver/src/dns_lru.rs

```diff
@@ -80,116 +81,165 @@ impl LruValue {
 /// An LRU eviction cache specifically for storing DNS records
 #[derive(Clone, Debug)]
 pub struct DnsLru {
-    cache: Arc<Mutex<LruCache<Query, LruValue>>>,
-    /// A minimum TTL value for positive responses.
-    ///
-    /// Positive responses with TTLs under `positive_min_ttl` will use
-    /// `positive_min_ttl` instead.
-    ///
-    /// If this value is not set on the `TtlConfig` used to construct this
-    /// `DnsLru`, it will default to 0.
-    positive_min_ttl: Duration,
-    /// A minimum TTL value for negative (`NXDOMAIN`) responses.
-    ///
-    /// `NXDOMAIN` responses with TTLs under `negative_min_ttl` will use
-    /// `negative_min_ttl` instead.
-    ///
-    /// If this value is not set on the `TtlConfig` used to construct this
-    /// `DnsLru`, it will default to 0.
-    negative_min_ttl: Duration,
-    /// A maximum TTL value for positive responses.
-    ///
-    /// Positive responses with TTLs over `positive_max_ttl` will use
-    /// `positive_max_ttl` instead.
-    ///
-    ///  If this value is not set on the `TtlConfig` used to construct this
-    /// `DnsLru`, it will default to [`MAX_TTL`] seconds.
-    ///
-    /// [`MAX_TTL`]: const.MAX_TTL.html
-    positive_max_ttl: Duration,
-    /// A maximum TTL value for negative (`NXDOMAIN`) responses.
+    inner: Arc<DnsLruInner>,
```

Comment:
Would be good to split out the interjection of `DnsLruInner` into a separate commit.